inference-engine/src/transformations/src/transformations/common_optimizations/log_softmax_fusion.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/log_softmax_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::LogSoftmaxFusion, "LogSoftmaxFusion", 0);
+
+ngraph::pass::LogSoftmaxFusion::LogSoftmaxFusion() {
+    auto input = ngraph::pattern::any_input();
+    auto axis1 = ngraph::pattern::wrap_type<ngraph::opset5::Constant>();
+    auto axis2 = ngraph::pattern::wrap_type<ngraph::opset5::Constant>();
+    auto max = std::make_shared<ngraph::opset5::ReduceMax>(input, axis1, true);
+    auto sub = std::make_shared<ngraph::opset5::Subtract>(input, max);
+    auto exp = std::make_shared<ngraph::opset5::Exp>(sub);
+    auto sum = std::make_shared<ngraph::opset5::ReduceSum>(exp, axis2, true);
+    auto log = std::make_shared<ngraph::opset5::Log>(sum);
+    auto sub_end = std::make_shared<ngraph::opset5::Subtract>(sub, log);
```

Comment:
Do we really have such a case of model where `LogSoftMax` is presented in this way? I understand decomposition but fusion seems very unlikely.

Reply:
MO was producing such subgraph before.

Reply:
I see. It overloads pattern matcher and we always require customers to regenerate their IRs for performance gain.

Reply:
I think that such pattern fusing is a nice have, I tried it on IRs generated with LogSoftmax decomposition, but maybe @lazarevevgeny or @ilyachur have a different opinions, especially from a binary size perspective.